### PR TITLE
RD-1983 add support for topics and keyphrases

### PR DESCRIPTION
**Pull request by iredpath**

### CODE REVIEW

File: model/src/main/java/com/basistech/rosette/dm/Topic.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                .add("confidence", confidence)
+                .add("conceptId", conceptId);
+    }
+
+    public static class Builder extends BaseAttribute.Builder<Topic, Topic.Builder> {
+        protected String concept;
+        protected Double confidence;
+        protected String conceptId;
+
+        public Builder(String concept, Double confidence, String conceptId) {
```
Comment (Katsuya-Tomioka) [inline model/src/main/java/com/basistech/rosette/dm/Topic.java]: Seems you want a default constructor (or minimum requirement only) also, or you can skip other methods..?

File: model/src/main/java/com/basistech/rosette/dm/Topic.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import java.io.Serializable;
+import java.util.Map;
+
+public class Topic extends BaseAttribute implements Serializable {
+    private static final long serialVersionUID = 222L;
+    private final String concept;
+    private final Double confidence;
+    private final String conceptId;
+
+    public Topic() throws Exception {
```
Comment (iredpath) [inline model/src/main/java/com/basistech/rosette/dm/Topic.java]: whoops, this wasn't meant to make it in.  Will remove

Comment (Katsuya-Tomioka) [inline model/src/main/java/com/basistech/rosette/dm/Topic.java]: :)

File: json/src/main/java/com/basistech/rosette/dm/jackson/ConceptMixin.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.Map;
+
+/**
+ * {@link Concept}
+ */
+public abstract class ConceptMixin {
+    @JsonCreator
+    ConceptMixin(@JsonProperty("concept") String concept,
+                 @JsonProperty("salience") Double salience,
+                 @JsonProperty("conceptId") String topicId,
```
Comment (guruxu) [inline json/src/main/java/com/basistech/rosette/dm/jackson/ConceptMixin.java]: is this intentionally different between json and java?

Comment (iredpath) [inline json/src/main/java/com/basistech/rosette/dm/jackson/ConceptMixin.java]: nope, just missed in the big concept <-> topic switch.  pushed a fix

**Pull request merged**